src/vorta/views/repo_add_dialog.py

```diff
@@ -196,43 +132,80 @@ def validate(self):
 
         return True
 
-    def password_listener(self):
+
+class AddRepoWindow(RepoWindow):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.passwordInput = PasswordInput()
+        self.passwordInput.add_form_to_layout(self.repoDataFormLayout)
+
+        self.encryptionComboBox.activated.connect(self.display_backend_warning)
+        self.encryptionComboBox.currentIndexChanged.connect(self.encryption_listener)
+
+        self.display_backend_warning()
+
+    def set_password(self, URL):
+        '''Autofill password from keyring only if current entry is empty'''
+        password = VortaKeyring.get_keyring().get_password('vorta-repo', URL)
+        if password and self.passwordInput.get_password() == "":
+            self.passwordInput.set_error_label(self.tr("Autofilled password from password manager."))
+            self.passwordInput.passwordLineEdit.setText(password)
+            self.passwordInput.confirmLineEdit.setText(password)
+
+    @property
+    def values(self):
+        out = dict(
+            ssh_key=self.sshComboBox.currentData(),
+            repo_url=self.repoURL.text(),
+            password=self.passwordInput.get_password(),
+            extra_borg_arguments=self.extraBorgArgumentsLineEdit.text(),
+            encryption=self.encryptionComboBox.currentData(),
+        )
+        return out
+
+    def encryption_listener(self):
         '''Validates passwords only if its going to be used'''
         if self.values['encryption'] == 'none':
-            self.passwordLabel.setText("")
-            return True
+            self.passwordInput.set_validation_enabled(False)
         else:
-            firstPass = self.passwordLineEdit.text()
-            secondPass = self.confirmLineEdit.text()
-            msg = validate_passwords(firstPass, secondPass)
-            self.passwordLabel.setText(translate('utils', msg))
-            return not bool(msg)
+            self.passwordInput.set_validation_enabled(True)
+
+    def display_backend_warning(self):
+        '''Display password backend message based off current keyring'''
+        if self.encryptionComboBox.currentData() != 'none':
+            self.passwordInput.set_error_label(VortaKeyring.get_keyring().get_backend_warning())
+
+    def run(self):
+        if self.validate() and self.passwordInput.validate():
```

Comment:
Wouldn't it make sense to include that check in `self.validate()`?

Reply:
I didn't add it because `passwordInput.validate()` is not available in `ExistingRepoWindow`. I can either a conditional check for type of passwordInput in base class to call it or implement a `validate()` method in `PasswordLineEdit` if you want to move it to `validate()`